Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -310,42 +311,46 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
         m_cachedCurrentTime = std::nullopt;
 }
 
-void AnimationTimelinesController::registerNamedScrollTimeline(const AtomString& name, Element& source, ScrollAxis axis)
+static Element* originatingElement(Ref<AnimationTimeline> timeline)
 {
-    auto it = m_nameToScrollTimelineMap.find(name);
-    if (it != m_nameToScrollTimelineMap.end()) {
-        auto& existingScrollTimeline = it->value;
-        existingScrollTimeline->setSource(&source);
-        existingScrollTimeline->setAxis(axis);
-    } else {
-        auto newScrollTimeline = ScrollTimeline::create(name, axis);
-        newScrollTimeline->setSource(&source);
-        m_nameToScrollTimelineMap.set(name, WTFMove(newScrollTimeline));
-    }
+    return is<ViewTimeline>(timeline) ? downcast<ViewTimeline>(timeline)->subject() : downcast<ScrollTimeline>(timeline)->source();
 }
 
-void AnimationTimelinesController::unregisterNamedScrollTimeline(const AtomString& name)
+Vector<Ref<AnimationTimeline>>& AnimationTimelinesController::timelinesForName(const AtomString& name)
 {
-    m_nameToScrollTimelineMap.remove(name);
+    return m_nameToTimelineMap.ensure(name, [] {
+        return Vector<Ref<AnimationTimeline>> { };
+    }).iterator->value;
 }
 
-ScrollTimeline* AnimationTimelinesController::scrollTimelineForName(const AtomString& name) const
+void AnimationTimelinesController::registerNamedScrollTimeline(const AtomString& name, const Element& source, ScrollAxis axis)
 {
-    return m_nameToScrollTimelineMap.get(name);
+    auto& timelines = timelinesForName(name);
+
+    auto existingTimelineIndex = timelines.findIf([&](auto& timeline) {
+        return is<ScrollTimeline>(timeline) && originatingElement(timeline) == &source;
+    });
```

Comment:
Shouldn't this check this is a `ScrollTimeline` and _not_ a `ViewTimeline`? Once we only store `ScrollTimeline` you can only check that it's not a `ViewTimeline`.

Reply:
Fixed.